rust/serving/src/source.rs

```diff
@@ -98,24 +109,32 @@ impl ServingSourceActor {
                 let _ = reply_to.send(messages);
             }
             ActorMessage::Ack { offsets, reply_to } => {
-                self.ack(offsets).await;
-                let _ = reply_to.send(());
+                let status = self.ack(offsets).await;
+                let _ = reply_to.send(status);
             }
         }
     }
 
-    async fn read(&mut self, count: usize, timeout_at: Instant) -> Vec<Message> {
+    async fn read(&mut self, count: usize, timeout_at: Instant) -> Result<Vec<Message>> {
         let mut messages = vec![];
         loop {
+            // Stop if the read timeout has reached or if we have collected the requested number of messages
             if messages.len() >= count || Instant::now() >= timeout_at {
                 break;
```

Comment:
Can we use `ReceiverStream` instead of `mpsc` channel to store the messages? That way you can chunk the stream using `chunks_timeout()`?

Reply:
I can't use `ReceiverStream` here since `chunks_timeout` consumes `self`. This results in:
```
cannot move out of `self.messages` which is behind a mutable reference
```